components/server/src/main/java/com/hotels/styx/server/netty/codec/NettyToStyxRequestDecoder.java

```diff
@@ -80,10 +76,11 @@ protected void decode(ChannelHandlerContext ctx, HttpObject httpObject, List<Obj
 
         try {
             if (httpObject instanceof HttpRequest) {
-                this.producer = new FlowControllingHttpContentProducer(ctx, this.flowControlEnabled);
-                Observable<ByteBuf> contentObservable = Observable.create(contentSubscriber -> {
-                    contentSubscriber.setProducer(this.producer);
-                    this.producer.subscriptionStart(contentSubscriber);
+                this.producer = new FlowControllingHttpContentProducer(ctx);
+                Flux<ByteBuf> contentObservable = Flux.create(sink -> {
+                    ctx.channel().config().setAutoRead(false);
```

Comment:
Hi @OwenLindsell We need to set `autoRead(false)` straight away after receiving the headers. That is, move it out of the subscription. Otherwise the flow control only kicks later when the Styx "client side" subscribes to the request content. This might be a lot later if should there be any asynchronous interceptors in the forwarding path.

Reply:
good spot.  I've moved it 👍